core/DataAccess/ArchiveTableDao.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\DataAccess;
+
+use Piwik\Config;
+use Piwik\Db;
+
+/**
+ * Data Access class for querying numeric & blob archive tables.
+ */
+class ArchiveTableDao
+{
+    /**
+     * @var string
+     */
+    private $tablesPrefix;
+
+    public function __construct(Config $config)
+    {
+        $this->tablesPrefix = $config->database['tables_prefix'];
+    }
+
+    /**
+     * Analyzes numeric & blob tables for a single table date (ie, `'2015_01'`) and returns
+     * statistics including:
+     *
+     * - number of archives present
+     * - number of invalidated archives
+     * - number of temporary archives
+     * - number of error archives
+     * - number of segment archives
+     * - number of numeric rows
+     * - number of blob rows
+     *
+     * @param string $tableDate ie `'2015_01'`
+     * @return array
+     */
+    public function getArchiveTableAnalysis($tableDate)
+    {
+        $numericQueryEmptyRow = array(
+            'count_archives' => '-',
+            'count_invalidated_archives' => '-',
+            'count_temporary_archives' => '-',
+            'count_error_archives' => '-',
+            'count_segment_archives' => '-',
+            'count_numeric_rows' => '-',
+        );
+
+        $tableDate = str_replace("`", "", $tableDate); // for sanity
+
+        $numericTable = $this->tablesPrefix . "archive_numeric_$tableDate";
+        $blobTable = $this->tablesPrefix . "archive_blob_$tableDate";
```

Comment:
Just trying to avoid the static method that will eventually have to be removed for DI. It's not important, just trying to instil a new habit.